src/Datagrid/ProxyQuery.php

```diff
@@ -15,12 +15,63 @@
 
 use Doctrine\Common\Collections\Criteria;
 use Doctrine\DBAL\Types\Type;
+use Doctrine\ORM\EntityManager;
 use Doctrine\ORM\Query;
 use Doctrine\ORM\QueryBuilder;
 use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
 
 /**
  * This class try to unify the query usage with Doctrine.
+ *
+ * @method Query\Expr    expr()
+ * @method QueryBuilder  setCacheable($cacheable)
```

Comment:
We should use the argument types exposed at `QueryBuilder` ([ref](https://github.com/doctrine/orm/blob/6138afdca9b78a75e162fecaf8be20fff99e15f8/lib/Doctrine/ORM/QueryBuilder.php#L184)).
```suggestion
 * @method QueryBuilder  setCacheable(bool $cacheable)
```

Reply:
Since it's casting in boolean, it theorically accept anything

![image](https://user-images.githubusercontent.com/9052536/77802464-36bf0c80-707b-11ea-8273-53af23ca20ca.png)

Reply:
I know. I'm asking to enforce the type declaration from our side based on the `@param` docblock.

Reply:
```
@method QueryBuilder  setCacheable(bool $cacheable)
```
Wouldn't this let you believe if you want to override the function that you have to type the param: 
```
public function setCacheable(bool $cacheable) { ... }
```
And wouldn't this let you believe that$cacheable is always a bool when it can in fact be everything ?

I understand your point, but I was thinking as better we should use the same type declaration that the QueryBuilder use, no ?

Reply:
It's fine. Let's save these updates for later :+1: